File: payments-api/app/routers/notifications.py
from __future__ import annotations

from fastapi import APIRouter, Depends
from app.core.security import AuthUser, require_user
from app.services.notifications import (
    count_unread,
    delete_notification,
    list_notifications,
    mark_read,
)

router = APIRouter(prefix="/notifications", tags=["notifications"])


@router.get("")
async def get_notifications(
    limit: int = 50,
    user: AuthUser = Depends(require_user),
) -> dict:
    items = list_notifications(user.id, limit=limit)
    unread = count_unread(user.id)
    return {"items": items, "unread": unread}


@router.post("/read-all")
async def read_all(user: AuthUser = Depends(require_user)) -> dict:
    mark_read(user.id)
    return {"ok": True}


@router.post("/{notif_id}/read")
async def read_one(notif_id: str, user: AuthUser = Depends(require_user)) -> dict:
    mark_read(user.id, notif_id)
    return {"ok": True}


@router.delete("")
async def delete_all(user: AuthUser = Depends(require_user)) -> dict:
    delete_notification(user.id)
    return {"ok": True}


@router.delete("/{notif_id}")
async def delete_one(notif_id: str, user: AuthUser = Depends(require_user)) -> dict:
    delete_notification(user.id, notif_id)
    return {"ok": True}

File: frontend/src/api/notifications.js
import { api } from "./client";

export const getNotifications = async (limit = 50) => {
  const res = await api.get(`/notifications?limit=${limit}`);
  return res.data;
};

export const markAllRead = async () => {
  const res = await api.post("/notifications/read-all");
  return res.data;
};

export const markOneRead = async (id) => {
  const res = await api.post(`/notifications/${id}/read`);
  return res.data;
};

export const deleteNotification = async (id) => {
  const res = await api.delete(`/notifications/${id}`);
  return res.data;
};

export const deleteAllNotifications = async () => {
  const res = await api.delete("/notifications");
  return res.data;
};
